sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Receipt.cs

```diff
@@ -0,0 +1,216 @@
+            {
+                foreach (FormField itemField in itemsField.Value.AsList())
+                {
+                    if (itemField.Value.Type == FieldValueType.Dictionary)
+                    {
+                        IReadOnlyDictionary<string, FormField> itemPropertyFields = itemField.Value.AsDictionary();
+
+                        FormField<string> name = ConvertStringField("Name", itemPropertyFields);
+                        FormField<float> quantity = ConvertFloatField("Quantity", itemPropertyFields);
+                        FormField<float> price = ConvertFloatField("Price", itemPropertyFields);
+                        FormField<float> totalPrice = ConvertFloatField("TotalPrice", itemPropertyFields);
+
+                        items.Add(new ReceiptItem(name, quantity, price, totalPrice));
+                    }
+                }
+            }
+
+            return items;
+        }
+
+        /// <summary>
+        /// Converts a weakly-typed <see cref="FormField"/> into a strongly-typed <see cref="FormField{T}"/> of type
+        /// <c>string</c>. Returns <c>null</c> if field does not exist or if conversion is not possible.
+        /// </summary>
+        /// <param name="fieldName">The key to retrieve the field to be converted from the <paramref name="fields"/> dictionary.</param>
+        /// <param name="fields">The dictionary of fields in the recognized receipt. Returned by the service.</param>
+        /// <returns>A strongly-typed <see cref="FormField{T}"/>. Returns <c>null</c> if field does not exist or if conversion is not possible.</returns>
+        private static FormField<string> ConvertStringField(string fieldName, IReadOnlyDictionary<string, FormField> fields)
+        {
+            if (fields.TryGetValue(fieldName, out FormField field)
+                && field.Value.Type == FieldValueType.String)
+            {
+                return new FormField<string>(field, field.Value.AsString());
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Converts a weakly-typed <see cref="FormField"/> into a strongly-typed <see cref="FormField{T}"/> of type
+        /// <c>string</c>, but formatted as a phone number. Returns <c>null</c> if field does not exist or if conversion
+        /// is not possible.
+        /// </summary>
+        /// <param name="fieldName">The key to retrieve the field to be converted from the <paramref name="fields"/> dictionary.</param>
+        /// <param name="fields">The dictionary of fields in the recognized receipt. Returned by the service.</param>
+        /// <returns>A strongly-typed <see cref="FormField{T}"/>. Returns <c>null</c> if field does not exist or if conversion is not possible.</returns>
+        private static FormField<string> ConvertPhoneNumberField(string fieldName, IReadOnlyDictionary<string, FormField> fields)
+        {
+            if (fields.TryGetValue(fieldName, out FormField field)
```

Comment:
I wonder if we need to check `&& field.Value.Type == FieldValueType.PhoneNumber`. Each `asXX` method has already that validation

Reply:
You're not wrong, but the `asXX` methods throw if the type is incorrect. We would need to wrap the calls in `try/catch` blocks. Adding an extra condition to the `if` feels simpler.